tezos/messages/src/p2p/encoding/operations_for_blocks.rs

```diff
@@ -1,19 +1,28 @@
 // Copyright (c) SimpleStaking and Tezedge Contributors
 // SPDX-License-Identifier: MIT
 
-use std::mem::size_of;
+use std::collections::VecDeque;
 use std::sync::Arc;
 
+use bytes::Buf;
+
 use getset::{CopyGetters, Getters};
-use serde::{Deserialize, Serialize};
+use serde::{Deserialize, Serialize, Serializer};
 
 use crypto::hash::{BlockHash, Hash, HashType};
-use tezos_encoding::encoding::{Encoding, Field, HasEncoding, Tag, TagMap};
-use tezos_encoding::has_encoding;
+use tezos_encoding::binary_reader::BinaryReaderError;
+use tezos_encoding::encoding::{CustomCodec, Encoding, Field, HasEncoding};
+use tezos_encoding::ser::Error;
+use tezos_encoding::types::{RecursiveDataSize, Value};
+use tezos_encoding::{has_encoding, safe};
 
 use crate::cached_data;
 use crate::p2p::binary_message::cache::BinaryDataCache;
 use crate::p2p::encoding::operation::Operation;
+use tezos_encoding::json_writer::JsonWriter;
+
+/// Maximal depth for path to be decoded
+pub const PATH_MAX_DEPTH: RecursiveDataSize = 512;
```

Comment:
@akoptelov 
please add there detailed comments, why this is 512 (like we talked on slack)
\and TODO: with jira linked, where we will continue on that

then please rebase and merge